cli/src/api/pcluster_api.py

```diff
@@ -80,34 +102,50 @@ def delete_cluster(cluster_name: str, region: str, keep_logs: bool = True):
             # retrieve cluster config and generate model
             cluster = Cluster(cluster_name)
             cluster.delete(keep_logs)
-            return ClusterInfo(cluster)
-        except (ClusterActionError, Exception) as e:
-            return ApiFailure(e)
+            return FullClusterInfo(cluster)
+        except Exception as e:
+            return ApiFailure(str(e))
 
     @staticmethod
     def describe_cluster(cluster_name: str, region: str):
         """Get cluster information."""
         try:
             if region:
                 os.environ["AWS_DEFAULT_REGION"] = region
-            return ClusterInfo(Cluster(cluster_name))
-        except (ClusterActionError, Exception) as e:
-            return ApiFailure(e)
+            return FullClusterInfo(Cluster(cluster_name))
+        except Exception as e:
+            return ApiFailure(str(e))
 
     @staticmethod
     def update_cluster(cluster_name: str, region: str):
         """Update existing cluster."""
-        if region:
-            os.environ["AWS_DEFAULT_REGION"] = region
-        # Check if stack version matches with running version.
-        cluster = Cluster(cluster_name)
-
-        installed_version = get_installed_version()
-        if cluster.stack.version != installed_version:
-            raise ClusterActionError(
-                f"The cluster was created with a different version of "
-                f"ParallelCluster: {cluster.stack.version}. Installed version is {installed_version}. "
-                "This operation may only be performed using the same ParallelCluster "
-                "version used to create the cluster."
-            )
-        return ClusterInfo(cluster)
+        try:
+            if region:
+                os.environ["AWS_DEFAULT_REGION"] = region
+            # Check if stack version matches with running version.
+            cluster = Cluster(cluster_name)
+
+            installed_version = get_installed_version()
+            if cluster.stack.version != installed_version:
+                raise ClusterActionError(
+                    "The cluster was created with a different version of "
+                    f"ParallelCluster: {cluster.stack.version}. Installed version is {installed_version}. "
+                    "This operation may only be performed using the same ParallelCluster "
+                    "version used to create the cluster."
+                )
+            return FullClusterInfo(cluster)
+        except Exception as e:
+            return ApiFailure(str(e))
+
+    @staticmethod
+    def list_clusters(region: str):
+        """List existing clusters."""
+        try:
+            if region:
```

Comment:
If the region is not specified by command, which region will AWSApi choose during initialization?

Reply:
AWSApi uses boto3 so it will use the boto3 logic to retrieve the region to use: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#guide-configuration

Reply:
BTW in my patch I changed all the api commands to make the region a mandatory parameter.